### ISSUE TITLE
Question about tokens

### ISSUE THREAD

**Issue by nemo0526** [ninatu/everything_at_once#11]
Hello,what is your attention_mask and special_token_mask for?

**Comment by ninatu** [comment]
Hi!

Since the number of tokens may vary for different elements in a batch, for example, depending on the length of the video clip or the number of words, we create an attention mask for each element in a batch to indicate which tokens should not be used in self-attention. As I remember, special_token_mask is redundant and not used.

**Comment by nemo0526** [comment]
Okay, thanks for your reply!
